pandas/compat/_optional.py

```diff
@@ -93,10 +100,16 @@ def import_optional_dependency(
             raise ImportError(msg) from None
         else:
             return None
-
-    minimum_version = VERSIONS.get(name)
+    # Handle submodules: if we have submodule, grab parent module from sys.modules
```

Comment:
why is all this needed?

Reply:
This has been answered before: https://github.com/pandas-dev/pandas/pull/31817#discussion_r435181102 (and the above comment has been added based on your comment)

It's to 1) import a submodule (`pyarrow.csv` in this case) and 2) to support passing a different version as in our global minimum versions dictionary.

Now I suppose that the submodule importing is not necessarily needed. Right now this PR does:

```
csv = import_optional_dependency("pyarrow.csv", min_version="0.15")
```

but I suppose this could also be:

```
import_optional_dependency("pyarrow", min_version="0.15")
from pyarrow import csv
```

And then this additional code to directly import a submodule with `import_optional_dependency` is not needed (although where it is used, I think it is a bit cleaner to be able to directly import the submodule)

Reply:
@jorisvandenbossche importing as a submodule is required, you can't access the csv module by doing ``pyarrow.csv`` as far as I remember.